cmd/disk-uploader/main.go

```diff
@@ -0,0 +1,163 @@
+	imagePushTimeout := opts.pushTimeout
+
+	log.Printf("Creating a new Secret '%s/%s' object...", namespace, name)
+
+	if err := secrets.CreateVirtualMachineExportSecret(k8sClient, virtClient, namespace, name); err != nil {
+		return err
+	}
+
+	log.Printf("Creating a new VirtualMachineExport '%s/%s' object...", namespace, name)
+
+	if err := vmexport.CreateVirtualMachineExport(k8sClient, virtClient, kind, namespace, name); err != nil {
+		return err
+	}
+
+	log.Println("Waiting for VirtualMachineExport status to be ready...")
+
+	if err := vmexport.WaitUntilVirtualMachineExportReady(virtClient, namespace, name); err != nil {
+		return err
+	}
+
+	log.Println("Getting raw disk URL from the VirtualMachineExport object status...")
+
+	rawDiskUrl, err := vmexport.GetRawDiskUrlFromVolumes(virtClient, namespace, name, volumeName)
+	if err != nil {
+		return err
+	}
+
+	log.Println("Creating TLS certificate file from the VirtualMachineExport object status...")
+
+	certificateData, err := certificate.GetCertificateFromVirtualMachineExport(virtClient, namespace, name)
+	if err != nil {
+		return err
+	}
+
+	if err := certificate.CreateCertificateFile(certificatePath, certificateData); err != nil {
+		return err
+	}
+
+	log.Println("Getting export token from the Secret object...")
+
+	kvExportToken, err := secrets.GetTokenFromVirtualMachineExportSecret(virtClient, namespace, name)
+	if err != nil {
+		return err
+	}
+
+	log.Println("Downloading disk image from the VirtualMachineExport server...")
+
+	if err := disk.DownloadDiskImageFromURL(rawDiskUrl, kvExportTokenHeader, kvExportToken, certificatePath, diskPath); err != nil {
+		return err
+	}
+
+	log.Println("Building a new container image...")
+
+	containerImage, err := image.Build(diskPath)
+	if err != nil {
+		return err
+	}
+
+	log.Println("Pushing new container image to the container registry...")
+
+	if err := image.Push(containerImage, imageDestination, imagePushTimeout); err != nil {
+		return err
+	}
+
+	log.Println("Successfully uploaded to the container registry.")
+	return nil
+}
+
+func main() {
+	var opts RunOptions
+	var command = &cobra.Command{
+		Use:   "kubevirt-disk-uploader",
+		Short: "Extracts disk and uploads it to a container registry",
+		Run: func(cmd *cobra.Command, args []string) {
+			config, err := rest.InClusterConfig()
+			if err != nil {
+				log.Panicln(err)
+			}
+
+			k8sClient, err := k8sv1.NewForConfig(config)
+			if err != nil {
+				log.Panicln(err)
+			}
+
+			virtClient, err := kubecli.GetKubevirtClient()
+			if err != nil {
+				log.Panicln(err)
+			}
+
+			opts.k8sClient = k8sClient
+			opts.virtClient = virtClient
+
+			namespace := os.Getenv("POD_NAMESPACE")
+			if namespace != "" {
+				opts.exportSourceNamespace = namespace
+			}
+
+			if err := run(opts); err != nil {
+				log.Panicln(err)
+			}
+		},
+	}
+
+	command.Flags().StringVar(&opts.exportSourceKind, "export-source-kind", "", "specify the export source kind (vm, vmsnapshot, pvc)")
```

Comment:
Would you please align with param parsing as other modules are doing?
https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/modules/create-vm/pkg/utils/parse/clioptions.go#L25
Then the parsing is called from main e.g. https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/cmd/create-vm/main.go#L23